staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -182,16 +190,20 @@ func (s *store) Create(ctx context.Context, key string, obj, out runtime.Object,
 		clientv3.OpPut(key, string(newData), opts...),
 	).Commit()
 	metrics.RecordEtcdRequestLatency("create", getTypeName(obj), startTime)
+	trace.Step("Txn call finished", utiltrace.Field{"err", err})
 	if err != nil {
 		return err
 	}
+
 	if !txnResp.Succeeded {
 		return storage.NewKeyExistsError(key, 0)
 	}
 
 	if out != nil {
 		putResp := txnResp.Responses[0].GetResponsePut()
-		return decode(s.codec, s.versioner, data, out, putResp.Header.Revision)
+		err = decode(s.codec, s.versioner, data, out, putResp.Header.Revision)
+		trace.Step("decode finished", utiltrace.Field{"len", len(data)}, utiltrace.Field{"err", err})
```

Comment:
Actually - this doesn't seem needed.
The next line is return anyway - so this will effectively be equivalent with END step and "Step" adds an unnecessary memory allocation then.

Same for update

Reply:
Relying on END is problematic for at least two reasons:
* the name END doesn't provide much information to someone who isn't familiar with this code.
* the END is used not only here, but also in all other places where we leave this function (e.g. as part of error). In the past I remember at least few times when I was confused by this (I thought that part after last "Step" was slow, but it turned out that we errored before we even touched that part of code).

I think that this additional allocation is worth simplifying debugging in this area.

Reply:
OK - I can live with it.